partner_firstname/views/res_partner.xml

```diff
@@ -1,98 +1,85 @@
+            <field name="name">Add firstname and surnames</field>
+            <field name="model">res.partner</field>
+            <field name="inherit_id" ref="base.view_partner_form"/>
+            <field name="arch" type="xml">
+                <data>
+                    <xpath expr="//field[@name='name']" position="attributes">
+                        <attribute name="attrs">{
+                            'readonly': [('is_company', '=', False)],
+                            'required': [('is_company', '=', True)]
+                            }</attribute>
+                    </xpath>
 
-            <xpath expr="//field[@name='category_id']" position="before">
-                <group attrs="{'invisible': [('is_company', '=', True)]}">
-                    <field name="lastname" attrs=
-                        "{'required': [('firstname', '=', False),
-                                       ('is_company', '=', False)]}"/>
-                    <field name="firstname" attrs=
-                        "{'required': [('lastname', '=', False),
-                                       ('is_company', '=', False)]}"/>
-                </group>
-            </xpath>
+                    <xpath expr="//field[@name='parent_id']/.." position="before">
+                        <div attrs="{'invisible': [('is_company', '=', True)]}">
+                            <label for="lastname" string="Lastname"/>
+                            <field name="lastname" attrs=
+                                "{'required': [('firstname', '=', False),
+                                ('is_company', '=', False)]}" class="oe_inline"/><br/>
+                            <label for="firstname" string="Firstname"/>
+                            <field name="firstname" attrs=
+                                "{'required': [('lastname', '=', False),
+                                ('is_company', '=', False)]}" class="oe_inline"/>
+                        </div>
+                    </xpath>
 
-            <!-- Modify inner contact form of child_ids -->
-            <xpath expr="//field[@name='child_ids']/form
-                         //field[@name='category_id']"
-                   position="before">
-                <group attrs="{'invisible': [('is_company', '=', True)]}">
-                    <field name="lastname" attrs=
-                        "{'required': [('firstname', '=', False),
-                                       ('is_company', '=', False)]}"/>
-                    <field name="firstname" attrs=
-                        "{'required': [('lastname', '=', False),
-                                       ('is_company', '=', False)]}"/>
-                </group>
-            </xpath>
+                    <!-- Modify inner contact form of child_ids -->
+                    <xpath expr="//field[@name='child_ids']/form
+                        //field[@name='name']"
+                        position="before">
+                        <field name="is_company"/>
```

Comment:
Can a contact be a company? If we want to display it, we may want to display "company_type" field as in main form view